### PR TITLE
fix: Add complementing caveats to test tools

### PR DESCRIPTION
**Pull request by hampuslavin**
## Description
- Adds complementing caveats to test tools
- Related to this PR: https://github.com/serverpod/serverpod/pull/2851

### CODE REVIEW

File: docs/06-concepts/18-testing/05-caveates-and-edge-cases.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+The following two cases are deviating from true transaction behavior:
+
+### Concurrent transactions
+
+Concurrent transactions are not supported with the default rollback database setting, see [this section](the-basics#rollback-database-configuration) for more information on how to test these scenarios.
+
+### Database exceptions that are quelled
+
+There is a specific case where the test tools behavior deviates from production behvaior. See example below:
```
Comment (SandPod) [inline docs/06-concepts/18-testing/05-caveates-and-edge-cases.md:15]:
```suggestion
There is a specific case where the test tools behavior deviates from production behavior. See example below:
```

File: docs/06-concepts/18-testing/05-caveates-and-edge-cases.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+});
+
+// ATTENTION: This will throw an exception in production
+// but not in the test tools.
+await transactionFuture;
+```
+
+In production, the transaction call will throw if any database exception happened during its execution, _even_ if the exception was first caught inside the transaction. However, in the test tools this will not throw an exception due to how the nested transactions are emulated.
+
+Please note that this case should be very rare as the above code doesn't really make sense.
```
Comment (SandPod) [inline docs/06-concepts/18-testing/05-caveates-and-edge-cases.md:35]: Suggestion: Remove "opinion" from the documentation.

Maybe we should motivate that writing query code like this is not considered best practice, and therefore, we expect the impact of this deviation to be small.

File: docs/06-concepts/18-testing/05-caveates-and-edge-cases.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# Caveats
+
+## Transactions
+
+As explained in [the basics](the-basics#seeding-the-database), the test tools are by default wrapping each test case in a transaction. That ensures  database calls and even further transactions are contained within that top level transaction. Since nested transactions are not natively supported by PostgreSQL, the test tools emulates this behaviour by using save points.
+
+The following two cases are deviating from true transaction behavior:
+
+### Concurrent transactions
```
Comment (SandPod) [inline docs/06-concepts/18-testing/05-caveates-and-edge-cases.md:9]: Question: We have taken a different pattern for the database exceptions being quelled compared to concurrent transactions. One is documented here and one is documented in "the basics". Should we document them the same way?

**Review by SandPod — COMMENTED** [review]
Question: Do we really need a caveats section or could this be embedded into another part of the test documentation? Or have we seen this pattern in some other framework?

Maybe these are test framework assumptions?

**Comment by hampuslavin** [comment]
> Question: Do we really need a caveats section or could this be embedded into another part of the test documentation? Or have we seen this pattern in some other framework?
> 
> Maybe these are test framework assumptions?

Hmm, yea maybe should scrap this section 🤔  My main idea behind this was just to keep the other pages simpler, since this is kind of an edge case. But maybe it doesn't make sense.

**Comment by hampuslavin** [comment]
@SandPod Tried a different approach and treated it the same as the concurrent transactions notice. Let me know what you think! :bulb

**Review by SandPod — APPROVED** [review]
LGTM! 👍
